pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -81,6 +81,7 @@ private static void setupHelixClusterIfNeeded(String helixClusterName, String zk
       configMap.put(DEFAULT_HYPERLOGLOG_LOG2M_KEY, Integer.toString(DEFAULT_HYPERLOGLOG_LOG2M));
       configMap.put(CommonConstants.Broker.CONFIG_OF_ENABLE_QUERY_LIMIT_OVERRIDE, Boolean.toString(false));
       admin.setConfig(configScope, configMap);
+      admin.close();
```

Comment:
You need to close it in both cases? (Move it out of the else block)

Reply:
Updated.